mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -119,6 +119,18 @@ final class EMLDAOptimizer extends LDAOptimizer {
       }
     }
 
+    // create term vertices for empty docs
+    val emptyDocTermVertices: RDD[(VertexId, TopicCounts)] = {
+      //empty documents
+      val emptyDocs = docs.filter { case (d: Long, tc: Vector) =>
+        tc.toBreeze.reduce(_+_) != BDV(0.0)
```

Comment:
agree! in between I realize I had bug with != . it should be == . Since we want empty docs and hence the norm has to be zero.